### ISSUE TITLE
Basic example throws TemplateCompilationException in ASP.NET 5 

### ISSUE THREAD

**Issue by faniereynders** [Antaris/RazorEngine#354]
Given the standard example as outlined on the [getting started page](https://antaris.github.io/RazorEngine/index.html):

```csharp
using RazorEngine;
using RazorEngine.Templating;

namespace ConsoleApp1
{
    public class Person
    {
        public string Name { get; set; }
    }
    class Program
    {
        static void Main(string[] args)
        {
            string template = "Hello @Model.Name, welcome to RazorEngine!";
            var result =
                Engine.Razor.RunCompile(template, "templateKey", typeof(Person), new Person { Name = "World" });
        }
    }
}
```

This throws a weird exception on a ASP.NET 5 console application - using the `1.0.0-rc1-update1` runtime. 

Details of the exception:
```
Errors while compiling a Template.
Please try the following to solve the situation:
  * If the problem is about missing/invalid references or multiple defines either try to load 
    the missing references manually (in the compiling appdomain!) or
    Specify your references manually by providing your own IReferenceResolver implementation.
    See https://antaris.github.io/RazorEngine/ReferenceResolver.html for details.
    Currently all references have to be available as files!
  * If you get 'class' does not contain a definition for 'member': 
        try another modelType (for example 'null' to make the model dynamic).
        NOTE: You CANNOT use typeof(dynamic) to make the model dynamic!
    Or try to use static instead of anonymous/dynamic types.
More details about the error:
 - warning: (19, 9) Filename specified for preprocessor directive is empty
	 - error: (8, 101) The type or namespace name 'ConsoleApp1' could not be found (are you missing a using directive or an assembly reference?)
Temporary files of the compilation can be found in (please delete the folder): C:\Users\fanie\AppData\Local\Temp\RazorEngine_tsegnqfo.tpq
The template we tried to compile is: 
------------- START -----------
Hello @Model.Name, welcome to RazorEngine!
------------- END -----------

The generated source code is: 
------------- START -----------
namespace CompiledRazorTemplates.Dynamic
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Threading.Tasks;

    public class RazorEngine_b8e7fed8cfb7425583f80136b710d51d : RazorEngine.Templating.TemplateBase<ConsoleApp1.Person>
    {
        
        public RazorEngine_b8e7fed8cfb7425583f80136b710d51d()
        {
        }

        #pragma warning disable 1998
        public override async Task Execute()
        {
            WriteLiteral("Hello ");
#line 1 ""
 Write(Model.Name);

#line default

            WriteLiteral(", welcome to RazorEngine!");
        }
        #pragma warning restore 1998
    }
}

------------- END -----------


List of loaded Assemblies:
C:\Windows\Microsoft.NET\Framework\v4.0.30319\mscorlib.dll
	Loaded Assembly: C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System\v4.0_4.0.0.0__b77a5c561934e089\System.dll
	Loaded Assembly: C:\Users\fanie\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\Microsoft.Dnx.Host.dll
	Loaded Assembly: C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Configuration\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.dll
	Loaded Assembly: C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Core\v4.0_4.0.0.0__b77a5c561934e089\System.Core.dll
	Loaded Assembly: C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xml\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.dll
	Loaded Assembly: C:\Users\fanie\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\Microsoft.Extensions.PlatformAbstractions.dll
	Loaded Assembly: C:\Users\fanie\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\Microsoft.Dnx.Loader.dll
	Loaded Assembly: C:\Users\fanie\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\Microsoft.Dnx.ApplicationHost.dll
	Loaded Assembly: C:\Users\fanie\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\Microsoft.Dnx.Runtime.dll
	Loaded Assembly: C:\Users\fanie\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\Microsoft.Dnx.Compilation.dll
	Loaded Assembly: C:\Users\fanie\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\Microsoft.Dnx.Compilation.Abstractions.dll
	Loaded Assembly: C:\Users\fanie\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\Microsoft.Dnx.Compilation.DesignTime.dll
	Loaded Assembly: C:\Users\fanie\.dnx\packages\RazorEngine\4.2.7-beta1\lib\net45\RazorEngine.dll
	Loaded Assembly: C:\WINDOWS\assembly\GAC_MSIL\Microsoft.VisualStudio.Debugger.Runtime\14.0.0.0__b03f5f7f11d50a3a\Microsoft.VisualStudio.Debugger.Runtime.dll
	Loaded Assembly: C:\Users\fanie\.dnx\packages\Microsoft.AspNet.Razor\4.0.0-beta7\lib\dnx451\Microsoft.AspNet.Razor.dll
	Loaded Assembly: C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\Microsoft.CSharp\v4.0_4.0.0.0__b03f5f7f11d50a3a\Microsoft.CSharp.dll
```

> As a side note: I've tried this exact example on a *traditional* ConsoleApplication and it is working as expected.

**Comment by faniereynders** [comment]
As an update:

Although updating the target framework to .NET 4.6, the error still surfaces, but I found that specifying `null` as the `modelType` works. When the type is explicitly set it still throws `The type or namespace name 'ConsoleApp1' could not be found...`

**Comment by devteamexpress** [comment]
I'm getting the same issue here, we are heavily dependent on razor and just upgraded our app to Asp 5. Can someone take a look at this please?

**Comment by oleggolovkov** [comment]
I had the same problem and found a somewhat clumsy solution:
1. Go to your project properties and enable 'Produce outputs on build'.
2. Build it and find your dll, should be something like '...artifacts\bin\MyCoolApp\Debug\dnx46\MyCoolApp.dll'
3. Now you need a custom IReferenceResolver, use it like this:
            var config = new TemplateServiceConfiguration();
            config.ReferenceResolver = new MyIReferenceResolver();
            var service = RazorEngineService.Create(config);            
            Engine.Razor = service;
4. MyIReferenceResolver is based on the sample from documentation, but also includes the previously built dll:
class MyIReferenceResolver : IReferenceResolver {    
        public IEnumerable<CompilerReference> GetReferences(TypeContext context, IEnumerable<CompilerReference> includeAssemblies) {
            IEnumerable<string> loadedAssemblies = (new UseCurrentAssembliesReferenceResolver())
                .GetReferences(context, includeAssemblies)
                .Select(r => r.GetFile());
            foreach (var a in loadedAssemblies) {
                yield return CompilerReference.From(a);
            }
            yield return CompilerReference.From(@"....\artifacts\bin\MyCoolApp\Debug\dnx46\MyCoolApp.dll"); 
        }
    }

Hope this helps someone

**Comment by matthid** [comment]
Duplicate of https://github.com/Antaris/RazorEngine/issues/322

**Comment by Rudauke** [comment]
Was this resolved? I'm using RazorEngine 3.9.0 and this problem occurs in my tests project.
